Added an option to disable namespace-metadata

shinigami-777 opened the pull request:
Fixes #11585 
Added option in `values.yaml` in extensions charts to disable the namspace-metadata jobs for helm based installations.
The `createNamespaceMetadataJob`  flag should be set to false from  cli to disable it. Disable if lack of privileges require doing it manually.

shinigami-777 commented:
Providing namespace-metadata disable option #11585

alpeb commented on jaeger/charts/linkerd-jaeger/templates/namespace-metadata-rbac.yaml:
```diff
@@ -13,7 +14,9 @@ metadata:
   name: namespace-metadata
   namespace: {{.Release.Namespace}}
 {{- include "partials.image-pull-secrets" .Values.imagePullSecrets }}
+{{-end}}
 ---
+{{- if not .Values.skipNamespaceMetadataJobs}}
```
It's more practical to wrap the entire file in this guard, instead of each resource.

shinigami-777 replied:
thanks for the suggestion. I will make changes to wrap the entire file.

alpeb commented on jaeger/charts/linkerd-jaeger/templates/namespace-metadata-rbac.yaml:
```diff
@@ -76,3 +83,4 @@ subjects:
 - kind: ServiceAccount
   name: namespace-metadata
   namespace: {{.Release.Namespace}}
+{{-end}}
```
Please make sure your changes don't remove the required newline character at the end of the file.

alpeb reviewed (changes requested):
Hi @shinigami-777 , thanks for the contribution. Please add a comment to this PR describing your change, and a reference to the issue it fixes. Also make sure you test your change before submitting, to validate it addresses the issue.

shinigami-777 commented on charts/linkerd-control-plane/values.yaml:
```diff
@@ -21,6 +21,8 @@ controllerLogFormat: plain
 # -- namespace to send control plane traces to
 controlPlaneTracingNamespace: linkerd-jaeger
+# -- adding an option to disble namespace-metadata jobs
+skipNamespaceMetadataJobs: false
```
Thanks for reviewing this PR @alpeb . I will put these entries in the extensions charts as well.

alpeb replied:
Please remove them.

alpeb reviewed (commented):
I've added a suggestion for a different naming of this new setting. After applying that, please run `./bin/helm-docs` to regenerate the helm values documentation in the charts README.md files.

alpeb commented on jaeger/charts/linkerd-jaeger/values.yaml:
```diff
@@ -32,6 +32,9 @@ enablePodAntiAffinity: false
 # for more information
 #nodeAffinity:
 
+# -- adding an option to disble namespace-metadata jobs
+skipNamespaceMetadataJobs: false
```
```suggestion
# -- Creates a Job that adds necessary metadata to the extension's namespace
# during install; disable if lack of privileges require doing this manually
createNamespaceMetadataJobs: true
```

shinigami-777 replied:
I have made the suggested changes and regenerated the helm values documentation in the charts README.md files.

alpeb reviewed (commented):
Thanks @shinigami-777 , this looks good to me. I'd like however to amend my previous comment on naming: can you rename the setting `createNamespaceMetadataJob`, given that it's just one job (per extension)? Sorry for the nit picking!

alpeb commented on jaeger/charts/linkerd-jaeger/templates/namespace-metadata-rbac.yaml:
```diff
@@ -76,3 +77,4 @@ subjects:
 - kind: ServiceAccount
   name: namespace-metadata
   namespace: {{.Release.Namespace}}
+{{-end}}
```
You require spaces here or else the template breaks, as you can see in the integration tests results.
```suggestion
{{- end }}
```

shinigami-777 replied:
Thanks  @alpeb . Added spaces between those end statements.

alpeb replied:
Thanks, can you address my comment about naming above? After that, this PR looks fine from my perspective.

shinigami-777 commented:
> Thanks @shinigami-777 , this looks good to me. I'd like however to amend my previous comment on naming: can you rename the setting `createNamespaceMetadataJob`, given that it's just one job (per extension)? Sorry for the nit picking!

Yes I have renamed the setting `createNamespaceMetadataJob` from `skipNamespaceMetadataJobs` in the extensions files.

shinigami-777 commented:
> > Yes I have renamed the setting `createNamespaceMetadataJob` from `skipNamespaceMetadataJobs` in the extensions files.
> 
> I still see `skipNamespaceMetadataJobs` in the diff...
I don't see the `skipNamespaceMetadataJobs` in the changed files. Can you please point me to where it is there.

> > Yes I have renamed the setting `createNamespaceMetadataJob` from `skipNamespaceMetadataJobs` in the extensions files.
> 
> I still see `skipNamespaceMetadataJobs` in the diff...

I can't see the `skipNamespaceMetadataJobs` in the changed files. Can you please point me to where it is.

alpeb commented:
I meant to say the setting should be `createNamespaceMetadataJob` instead of `createNamespaceMetadataJobs`.

shinigami-777 commented:
> I meant to say the setting should be `createNamespaceMetadataJob` instead of `createNamespaceMetadataJobs`.

Yes I have updated it to `createNamespaceMetadataJob` from `createNamespaceMetadataJobs`.

alpeb reviewed (approved):
Thanks @shinigami-777 , this looks good to me!

shinigami-777 commented:
> Thanks @shinigami-777 , this looks good to me!

Thanks @alpeb for the guidance.
